contrabandInventory Changes (Fixed)

AgentSmithRadio opened the pull request:
## About the PR
I've added contrabandInventory menus, the kind that you access by manipulating the "Manager" wire on vending machines, to nearly every vending machine SS14. This is the second attempt, the first pull request had a merge conflict that required me to redo this process. [You can see the previous PR here.](https://github.com/space-wizards/space-station-14/pull/32911)

## Why / Balance
contrabandInventory is a woefully under-explored mechanic in SS14, and I wanted to take advantage if the world-building, flavour and gameplay opportunities available from populating these secret menus.

To note, beyond securing the equipment, hacking a vending machine isn't hard once you locate the Manager Wire in any given round. There are a few goals I have in mind:

1. The items inside of the machines should be faithful to what those machines are.
2. The items inside of the machines should raise questions about the SS14 universe, the corporations behind these machines, and what happened on the space station before the crew arrived on disaster day.
3. None of these items should become meta. If a vending machine begs to be hacked every shift, the items inside are too powerful.
4. The items shouldn't be high-end prestige items, but they should be interesting.
5. The vending machine stocks should never invalidate the role of Cargo in station function.

Two major changes of note:
First, Changeling Sting has been removed from the Shamblers EMAG menu, and Syndi-Cakes has been removed from the GetMore EMAG menu, to move those items to their contrabandInventory menus.
Second, figurines have been placed in their revelevant Drobe machines.

My reasoning for adding figurines to the machines is as follows:

1. It removes a lot of weight from purchasing figurines from Cargo, which is a fairly frivolous purchase, and makes obtaining a full crew set much more reasonable.
2. It gives a reason for innocent people to trespass or seek permission to hack Drobes for roleplaying purposes. It makes every department a bit less secure, which is fun for antags and an added joy for Security.
3. It can allow us to return the figurine objective back to Thief by making obtaining figurines accessible by hacking a crap-ton of Drobes across the station. It adds an element to thief gameplay that is interesting and different, but isn't in itself illegal.
4. Toys are fun, seeing more figures is part of what makes Cog interesting in its design, I'd love to see them show up more often.
5. This makes the Cargo figurine bounty much less awful to accomplish by making it reasonable to complete for knowledgeable players.

## Technical details
I went through every vending machine .yml file, added a "contrabandInventory:" line if there wasn't one, then added each item for those menus. I kept formatting consistent with the currently .yml files.

## Media
[Here is a new and revised video](https://youtu.be/xI2oJzyzJ84?si=FFCIIzYQcvv6VMxH) of how to access the contrabandInventory menu, and showing every vending machine contrabandInventory menu. EMAG menus not included. Video uploaded to YouTube due to file size, video is approximately 5 minutes long. Alternatively you can look at the Files changed menu on the top bar of this PR, and see every list directly from the .yml files, highlighted for your convenience.

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.


## Breaking changes
Previous merge conflict resolved, and this shouldn't interfere with other PR's I've seen.

**Changelog**
:cl: AgentSmithRadio
- add: Added manager wire hacking menus to nearly all vending machines. Get hacking and see what's hidden!

IProduceWidgets commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/discount.yml:
```diff
@@ -8,3 +8,6 @@
     FoodSnackEnergy: 3
     CigPackMixed: 2
+  contrabandInventory:
+    FoodFrozenSandwich: 3
+    FoodFrozenSandwichStrawberry: 3
```
Id rather the ice cream be a chef/kitchen thing. Makes for good RP.

AgentSmithRadio replied:
My reasoning for adding these was to give the Discount Dan's machines something actually "worth getting" that's still ultimately just junk food. Perhaps the freezer section of the machine has been locked out of regular access due to a product recall, or faults in those units, and hacking it unlocks the partially-raided selection (0-3 of each sandwich spawn in these machines).

ChefVend does make sense to include these sandwiches instead. As of now for the update, the ChefVend contains 2x Moldy Bread, 1x Egg Carton (Broken, and 1 Donk-Pocket Box with 6 Donk Pockets. I've mostly just trolled chef and given an alternate food supply for when kitchen is unstaffed. Bringing it up to 5 items feels excessive, but if I did that I'd probably cut out the Moldy Bread to make it a cleaner 4.

If we moved this, what should we include in the Discount Dan's instead?

AgentSmithRadio replied:
I have moved the ice cream sandwiches as 2-ofs to the ChefVend.

IProduceWidgets commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/gib.yml:
```diff
@@ -8,6 +8,10 @@
     DrinkLemonLimeCan: 2
     DrinkFourteenLokoCan: 2
+  contrabandInventory:
+    ClothingNeckStethoscope: 2
+    Saw: 2
+    Tourniquet: 3
```
Not sure I understand why these are in the soda vendor.

AgentSmithRadio replied:
This is the Dr. Gibb machine. While most soda machines either receive a different soda (Dr. Gibb in PwrGame "Much Better Than Dr. Gibb" machines, or at least that's what they kept screaming at me during research and testing) or their full-sized bottle equivalents for the Cola/Soda and Space Mountain Wind machines, I wanted to try something more interesting for Dr. Gibb.

This is very much a field amputation kit, with a stethoscope for someone to throw on to pretend to be a doctor before doing it. While there is no amputation in the game, I wanted this to be somewhat jarring, that the seemingly innocent soda machine for some reason has these tools for amputation, with no anesthesia, wound dressing or anything else, and play into the shock value of seeing it for the first time. The reason to come back to this machine is the tourniquets, which is why their spawn is set to 3, so there's a higher chance that one will be there in a situation where you actually need it. 

tl;dr, field amputation kit in Dr. "tastes like malpractice" Gibb machine in horror game.

Alpaccalypse replied:
Sound legit to me. Not sure how good saw is as a melee weapon though, is it in any way OP?

IProduceWidgets commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/magivend.yml:
```diff
@@ -9,6 +9,9 @@
     ClothingOuterWizardViolet: 3
     ClothingShoesWizard: 9
+  contrabandInventory:
+    ToyFigurineWizard: 1
+    ToyFigurineWizardFake: 1
```
I wouldn't include the wizard ones, iirc the wizard ones are for a spell we don't have yet where people get turned into figurines.

AgentSmithRadio replied:
Figurines removed.

IProduceWidgets commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/medical.yml:
```diff
@@ -9,4 +9,5 @@
     ClothingEyesEyepatchHudMedical: 2
-
+  contrabandInventory:
+    Defibrillator: 1
```
Its probably not a good idea to have this many defibs, and having it in here makes it hard to balance for medbay.

thebadman4662 replied:
Defib is literally 3 steel, its way easier to get it printed most of the time, so it would be more of an emergency defib.

AgentSmithRadio replied:
thebadman4662 is right, the idea of this is that it's an emergency defibrillator that a Greytider/Salvager/Engineer can pop out of a NanoMed Plus during an emergency, where there's either power or they've gone to the lengths of finding a generator to power it. I'm actually still surprised with how cheap defibs are to make.

Defibs are cheap, but NanoMed Plus machines are common, and honestly the medium power cell inside is probably the most valuable part of it. I don't see people just going into medbay, hacking every NanoMed Plus and kicking out all of the defibs. However, in such a world I'd be fine adding singles of medicated sutures and regenerative mesh (1x stack of 1) as a "tease" item and as an accessible instructional tool for teaching medical interns. Otherwise, I need more ideas on what should go in here.

To note, my proposal has the WallMed/NanoMed machines getting 2x small power cell, specifically to charge a defib for one jolt in an emergency.

AgentSmithRadio replied:
Replaced NanoMed defibrillator for 1-of singles of medicated suture and regenerative mesh.

IProduceWidgets commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/starkist.yml:
```diff
@@ -8,6 +8,8 @@
     DrinkLemonLimeCan: 2
     DrinkFourteenLokoCan: 2
+  contrabandInventory:
+    FoodMealSashimi: 2
```
Probably too consistent considering is gives the very potent CarpoToxin

thebadman4662 replied:
Sashimi guarantees bunch of Opp for chemlab, I will defo be raiding kitchen every round if I am chem after it gets added... and restocks are only like 560$ each so thats very cheap roundstart mass cogni source as managament gets restocked too.

AgentSmithRadio replied:
To note, this is for the Star-Kisk Machine, it's not a consistent spawn on stations, and the Sashimi appears anywhere from 0-2.

The reasoning for the Sashimi is to deliberately have a fishy, but not Space Carp Fillet item to explain the Starkisk Can item description: "The taste of a star in liquid form. And, a bit of tuna...?" The soda only "tastes like orange soda", so I wanted this environmental bit to explain the fishiness, and it seemed like the funniest one to just find in a machine. 

When discussing this proposal on the SS14 discord, this has come up frequently, but no one has been able to explain to me how you get the Carpotoxin out of Sashimi short of extracting someone's blood after digesting it. Syringe and blender don't work. I've heard that the recycler and industrial reagent grinder can grind it up, but I can't get the recycler to break apart the sashimi on my private client and I've never once seen an industrial reagent grinder used in-game. Is there a verified way on how you actually accomplish this?

I'm also wondering if balance is actually affected given the current Salvage meta of doing Space Debris early, and frequently. An average roll debris is ~3 carp, which is 6 fillet if butchered on the spot, which is becoming a more common practice in my gameplay. 3 debris, 18 fillet, and it can easily high-roll even better than this. I'm not sure what a potential 4 fillet worth of Carpotoxin from a high roll sashimi spawn in a Starkisk machine is going to do here. That is to say, Carpotoxin doesn't currently feel rare when I play Salvager, or as Chemist, especially if you just ask Salvage to take the few seconds to farm it. It feels like the effort to farm a Starkisk machine for fillets, assuming it's currently possible, is more work than Salvage bothering to butcher Space Carp while they work.

In terms of substitutions, we can reduce yield by using the fillet-o-carp burger which has half the fillet yield, or just throw a space carp plushie in and find something else to go into the Salvage machine.

thebadman4662 replied:
> When discussing this proposal on the SS14 discord, this has come up frequently, but no one has been able to explain to me how you get the Carpotoxin out of Sashimi short of extracting someone's blood after digesting it.

Just drain it in chemlab/morgue or eat and vomit if you extremely concerned about wasting any unit of it and have enough ipecac.

AgentSmithRadio replied:
Sorry, what do you mean by drain it in the chemlab or morgue? 

And yeah, vomiting makes way more sense to me as a form of improvised extraction. I think that'd be an interesting mechanic, but yeah, I could see that being an issue.

AgentSmithRadio replied:
Oh crap, that actually works. I had no idea this was a thing with cooked foods: 
![image](https://github.com/user-attachments/assets/b63f022c-fa8b-4b64-875e-7195c47caac8)

To note, the Fillet-O-Carp burger doesn't contain Carpotoxin. I'll make the substitution, good catch:
![image](https://github.com/user-attachments/assets/df52813a-8005-4e52-aefc-4410a506e76b)

AgentSmithRadio replied:
Fillet-O-Carp is now in the machine instead of Sashimi.

thebadman4662 replied:
> Oh crap, that actually works. I had no idea this was a thing with cooked foods: ![image](https://private-user-images.githubusercontent.com/182301147/378466363-b63f022c-fa8b-4b64-875e-7195c47caac8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1NDk4MzQsIm5iZiI6MTczMDU0OTUzNCwicGF0aCI6Ii8xODIzMDExNDcvMzc4NDY2MzYzLWI2M2YwMjJjLWZhOGItNGI2NC04NzVlLTcxOTVjNDdjYWFjOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwMlQxMjEyMTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOWJjNWY2ZDVkYjU5ZmYyOWE2MzU0NTEzYWNlZGUxNWVhNWY3NTVmMTA5MzA3OGM5NDFjOTJjNzE3YmIzOGE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Niz6hWMb1iIqiDyIOtJTHa66_kmk3tOYj_zY87DehT0)
> 
> To note, the Fillet-O-Carp burger doesn't contain Carpotoxin. I'll make the substitution, good catch: ![image](https://private-user-images.githubusercontent.com/182301147/378467199-df52813a-8005-4e52-aefc-4410a506e76b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1NDk4MzQsIm5iZiI6MTczMDU0OTUzNCwicGF0aCI6Ii8xODIzMDExNDcvMzc4NDY3MTk5LWRmNTI4MTNhLTgwMDUtNGU1Mi1hZWZjLTQ0MTBhNTA2ZTc2Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwMlQxMjEyMTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NmU4ODUzOTc5YTJhMjJhM2NiMmMxYWJkMWQyMWU4NDNjOGRlN2RiMjk4NWM2OWYxMDM4YmMzNjBkYzhmYWRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4j6-Qw0jYDTrw2ig8KZGN375yS0tCgfvKBwyrl_NJKw)

So Lilten made new [video ](https://www.youtube.com/watch?v=tRWHrX4aIDk)and I realized the bug have evolved as you were not able to to drain food directly into chemmaster(it would only play sound)... Now that I think about it maybe jugs were not able to always drain too? Perhaps thats when it changed. I was using floor drains for no reason and dilluting plant chems to minimize the loss. x)

AgentSmithRadio replied:
Wow, I was not ready for that video, lol. Yeah, it's going to have implications, and I've already given Sashimi to Chemistry to drain out its Carpotoxin in one of my rounds.

I wonder how long the bug is going to last. Otherwise, now that I know it's _possible_, I'm going to avoid the possibility of it being easy chems like an Arnolds Pizza is.

IProduceWidgets commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/vendomat.yml:
```diff
@@ -10,3 +10,5 @@
     CapacitorStockPart: 4
     MicroManipulatorStockPart: 4
+  contrabandInventory:
+    KitchenKnife: 1
```
Scalpel would be funnier

AgentSmithRadio replied:
I'm not sure why it'd be funnier, but it serves the exact same purpose. And without looking it up, the damage difference is what, 7 to 8?

I kind of like this one, and it feels like a more relevant fit to the VendoMat, it feels more "techy" than the kitchen knife does. I'll likely just do that swap without further prompting.

thebadman4662 replied:
Scalpel is fast weapon so its actually 12 dps.

AgentSmithRadio replied:
Fair point. It's not amazing, I've had to defend myself as a Chemist using one on several occasions at this point, but that's good to know.

I still like this suggestion. I mostly just want to throw a "legitimate" cutting tool to the resourceful who can get their hands on a screwdriver and wirecutter, and are willing to risk getting shocked finding the manager wire that round.

AgentSmithRadio replied:
Kitchen Knife swapped for Scalpel.

AgentSmithRadio commented:
I apparently merge conflicted... myself? I'll get this fixed later today when I have more time to mess around in SmartGit.

RedBookcase commented:
An overall great PR that deserves to be merged. Currently the main interaction I have with the manager wire is using it for a free multitool if i'm tiding, or explaining how it works when teaching new players hacking. Without fail they always find the idea of a hidden inventory in a vending machine to be really cool, and then they're inevitably disappointed when they learn there's only about 3 machines with actual hidden inventories. Adding this gives new players actual rewards for being curious and fiddling around with existing machines, which I think is just a net positive. I don't think any of the items are especially powerful either (The most powerful being _maybe_ the extra mutagen in the botany vendor) so I don't think there's really much to worry about balance wise. Hope to see it in game soon!

TheDoctor1977 commented:
> That said, maybe 30u of hackable Mutagen is too accessible as a powerful poison for antagonists. I'm second-guessing myself now, I need to think a bit more on this one.

The botany chemistry kit from #31738 arguably contains far more potential for the smart antagonist, so I'd say it's probably fine.

AgentSmithRadio commented:
> The botany chemistry kit from #31738 arguably contains far more potential for the smart antagonist, so I'd say it's probably fine.

Fair point. After further thought, I'm not that worried. If this somehow becomes an issue it's easily removed anyways, and I don't see Botany not going to Chemistry for muta, it should be fine.

SlamBamActionman commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/seeds.yml:
```diff
@@ -39,5 +39,8 @@
     WheatSeeds: 5
     WatermelonSeeds: 5
+  contrabandInventory:
+    NettleSeeds: 2
```
Considering there have been talks about buffing deathnettle, I would not make it more available like this.

AgentSmithRadio replied:
I... did not know that. Amantia is pretty strong as it is, so probably not that. Off the top of my head, I should just swap this for Tobbaco or Bungo seeds, basically just one of the lesser-used exotics. Probably Bungo.

AgentSmithRadio replied:
Swapped for Bungo Seeds. While its pits can make a decent poison at .2u/sec at 1.8 poison/sec, it's much weaker than Fly Amantia and the upcoming Nettle, and literally the only other choice is tobacco. I think this fits what I was going for with Nettle.

SlamBamActionman commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/medical.yml:
```diff
@@ -10,4 +10,6 @@
     ClothingEyesEyepatchHudMedical: 2
-
+  contrabandInventory:
+    BrutepackAdvanced1: 1
```
Strong disagree with this, even just one will have a tider foaming at the mouth to get.

AgentSmithRadio replied:
Originally this was a defib before I realized how awful that would be, even when they only cost like, 3 steel. I replaced it with these 1-ofs of the advanced topicals as an emergency med you can hack for (with 3 NanoMed Plus per medbay, that's 3 of each), and as a way for doctors to teach interns about them and demonstrate their use.

I might just need to scrap this one all together.

AgentSmithRadio replied:
Swapped for a singular apple. I think the joke tells itself.

SlamBamActionman commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/cart.yml:
```diff
@@ -16,3 +16,6 @@
     EncryptionKeySecurity: 1
     EncryptionKeyService: 3
+  contrabandInventory:
+    RubberStampGreytide: 1
```
Meant to be a very rare maints item, and also weird that HoP would have it. Keep the pen, not the stamp.

AgentSmithRadio replied:
In my original design it was going to be a Syndicate Stamp, as sort of a trophy for being able to break into the HoP's office and hack their vending machine, but there was general protest in my discussions about this because those sell for 2TC.

I wanted there to be some sort of untraceable trophy in this machine for being able to pull it off, but perhaps it's too much.

SlamBamActionman replied:
As seen with the Sec clothesvend and the funny clown mask, the people who are most likely to hack open a vending machine are the people who have direct access to it. 

Maybe an Ian and Nanotrasen balloon, since it's in HoP's office?

AgentSmithRadio replied:
I dig it, good ideas there.

AgentSmithRadio replied:
Swapped the Greytider Stamp for 2 NT Balloons. I really like this one for the general vibe of what the HoP is going for.

AgentSmithRadio commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/vendomat.yml:
```diff
@@ -10,3 +10,5 @@
     CapacitorStockPart: 4
     MicroManipulatorStockPart: 4
+  contrabandInventory:
+    Scalpel: 1
```
I was hoping to provide a sort of legitimate knife-tool for passengers/greytiders. Originally it was a kitchen knife, after discussion here I swapped it to scalpel. I'm not sure where else to go on this front, there's not much else that makes sense in this machine.

AgentSmithRadio replied:
I think that the answer is a Network Configurator. It fits the theming of this machine, and its the natural companion of the Remote Signaller in there.

AgentSmithRadio replied:
Realized the Network Configurator is already stocked in the YouTool, I spaced on that one. Added a Proximity Sensor and Left Borg Arm to allow for a single partial bot creation without requiring Science. This allows for a Medibot, Cleanbot, Firebot and Honkbot without requiring further input from Science.

The worst case scenarios I came up with are round-removal inaprovaline kills using the Medibot, which are hard to pull off, and someone creating a Medibot off the grid and emagging it to cause some chaos on the station with sleepy-time drug injections. Both of these are already doable by stealing Medibots from medbay, so I'm not that concerned at this having a real balance impact.

AgentSmithRadio commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/wallmed.yml:
```diff
@@ -6,3 +6,5 @@
     EpinephrineChemistryBottle: 3
     Syringe: 3
+  contrabandInventory:
+    PowerCellSmall: 2
```
Yeah, if these were even medium cells they'd be hacked for every shift. They're something an engineer/tider can pop out during a medbay emergency whenever the batteries are injected with plasma, or they're all dead. This is one of the interactions I'm more excited to see out of this entire PR.

AgentSmithRadio commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/engivend.yml:
Otherwise undocumented note, I moved the Cow Tools from the YouTool to the EngiVend. I like the idea of them being accessible, but balance comments about providing a second semi-secret multitool to the YouTool is going a bit over the edge. Engineers will be pulling this out for a laugh moreso than anything else, or remember it's there whenever they hack this machine for the beer inside.

To note, the YouTool didn't get a replacement item. It already has the Multitool, it's pretty iconic in the current greytider meta, it's probably fine just leaving it as it is.

Alpaccalypse replied:
Beer for engivend is a nice touch.

AgentSmithRadio replied:
The visual that comes to mind for me is [the cutscene from the Starcraft 1 Zerg campaign](https://youtu.be/WI-utH8_D-Y?si=5IIizCnuk1AtJPcN) where a bunch of Dominion soldiers board one of their science vessels to look for signs of Zerg activity, only to pop open their nuclear bomb case to reveal a cooler with beer and ice in it.

I figure that throwing a few beers into the EngiVend is a good focal point for Engineering social get-togethers, say at the start of a shift, after setting up the singulo or when holded-in during a Zombie round. It just feels fitting that Engineers would tamper with their own vending machines, and if I were to do that, I'm definitely stuffing beer in there for later.

ps3moira commented:
Feels bloaty if you're just adding toy figurines

RedBookcase commented:
> This is bloat content if you're just adding toy figurines and plushies to every vending machine. Doesn't invoke world building or anything like that just makes it easier to do bounties for cargo

It adds a lot more than just figurines, you can watch the vid to see everything included. It makes the manager wire an actual functional part of every vending machine, rather than something that's useless on 95% of machines. And besides, I don't really see a problem with making figurines more obtainable.

AgentSmithRadio commented:
> This is bloat content if you're just adding toy figurines and plushies to every vending machine. Doesn't invoke world building or anything like that just makes it easier to do bounties for cargo

Paradise Station in SS13 took the same approach I came to, figurines go into their relevant drobes. The primary goal of this specific approach is to bring back the figurine objective for thief.

Figurines are fun, silly toy items, and they're inherently collectable. Before, thieves could roleplay wanting to collect figurines, raid the library and whatever mystery boxes spawned on the station, and would then have to meta themselves to cargo to order a bunch more to reach their goal of 13-15 figurines. Even if you brought cash, it only takes one cargo member to report you to become security's favourite person to search. It wasn't great.

If included, it wouldn't be uncommon for players of any job to start hacking drobes to fish these out. If this objective is brought back to thief, it gives them a legal way to obtain figurines, at least until they have to start trespassing departments to get their figurines, and possibly raise suspicion on themselves, especially if their collection is found. This loop reminds me most of the current lamp objective, but with a more common item type, and using hacking for machines instead of bypassing doors.

The core problem with designing manager wire menus is that you can't actually rock the balance all that strongly. Anything good enough to be hacked for in any given round will get hacked, we see that with the Multitool in YouTool machines every round on Wizden. I can tell stories with the amputation kit in the Dr. Gibb machine, or make jokes like apples in the NanoMed Plus, or sunflower seeds in the MegaSeed Servitor, but there's a fundamental limit. When I was developing this, I suggesting that a Syndicate Stamp get put into the HOP's vendor got ire because it costs TC, even though similar items like the Cybersun Pen, which costs 1TC, is not exclusive to the Uplink. 

People want small, so that's the direction I took this, everything is here for flavour in some way. Figurines are fun, a carp plushie in the Salvager machine is fun, but there's the functional like the Bottle of Nothing in the Booze-o-Mat, Bungo Seeds in the Servitor (as a poison and as a food), bot parts in the Vendomat, strange pills for the grinder lottery from the ChemVend, the box of Donk Pockets in the ChefVend (12u Omni per box). My goal was to make items good enough to be interesting, to be rewarding for someone exploring it for the first time, or someone going back because they know they can get a specific something, without actually breaking the current game balance. It's a minigame that now becomes more viable for people to explore, and acts as a gateway to more difficult hacking for new players (ie. doors in general, reversing EMAG, circumventing AI).

AgentSmithRadio commented:
> you can watch the vid to see everything included. 

My video is way out of date at this point, I've made too many changes and I think the video actually straight up forgot to check the condiment dispenser (which is a vending machine, so's the SmartFridge oddly enough). Short of me spending 40 minutes remaking the video, it's probably easier to check the Files Changed tab for an actual comprehensive list, unlike many PRs it's pretty easy to read and understand what the items are, even if you don't know the actual IDs.

Alpaccalypse commented:
Love the what your are doing with this PR. Even if you're on LRP this sort of stuff just adds to the game IMO. There's no point having contraband inventory if it doesn't exist 95% of the time you look for it, and your goals and balance for this seem spot on. Among other things, I was wondering why figurines even exist beyond hardcore RP content, this would answer that question. Someone mentioned bloat; this isn't bloat, it is making other game content no longer be bloat.

I can see why it is taking a while for this to get reviewed, compared to my vending related stuff. I've mostly been adding stuff that is mostly flavour or simply corrections/cleanup, and with a much more limited PR scope. This touches on a lot of the same areas of the game but with a subtle but tangible impact on gameplay.

I've not got any real constructive feedback to give, other then encouragment. The merit of what you have worked on with this speaks for itself. When I have a moment I'll have a proper look at your changes, and give you at least a non-maintainer review, as someone who had been working on the same .yml files for quite a while now.

Alpaccalypse commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/cigs.yml:
I mean come on, don't be such a square, surely you could have a chance of 1 spliff or some ground cannabis? Seems far more contrabandy than ground tobacco.

AgentSmithRadio replied:
Ground Tobbaco currently exists from growing and drying Tobacco in Botany (which isn't really ever done), or from buying the DIY Tobbaco Crate which has a bunch of it. This opens up that type of gameplay in a more accessible way, because let's face it, tobacco smokeables simply don't matter outside Soaked Smokes and Interdynes.

I currently have a joint in the NutriMax machine, as a way to keep cannabis purely in the realm of Botany, and as a way to show new players (who often do services jobs first, including Botany) the types of rewards they can get from working that profession.

Cigarette machines are currently very common, the average station gets what, 4, despite how random machine spawns work? If I add 4 ground cannabis, that's 4 joints I've just given to each station, which seems excessive, and takes agency away from Botany. I don't want to trample on their turf all that much.

Alpaccalypse replied:
Sounds reasonable. You've thought all this out well. Of course I can never find a machine when I actually grow cannabis and want some papers, but I reckon that is a skill issue. Now I mention it, what about adding rolling papers to the nutrimax? Either alongside the joint, or in place of it (as they are the source of cannabis for the most part).

I defer to how much thought you have given this though, don't let me pressure you to change what you think is already correct.

Alpaccalypse commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/medical.yml:
You could even make this a single golden apple, it is healing related, and doesn't seem too unbalanced as a chance to get if you hack.

AgentSmithRadio replied:
A single golden apple makes it possible to crack it, and give it to Botany for free golden apple seeds from the seed maker. It circumvents the mutation mechanic a bit too easily for an otherwise popular mutation. I find the idea that the apples are being deliberately locked away from doctors (the only people who have access to the machine) to be a funny enough joke, but the idea of providing a slight amount of TDD isn't completely off-base. 

If I really wanted to go this route, I could code in a 30u of TDD Bottle entity as the hack, but note that there's 3 NanoMed Plus machines in the average medbay, that's 90u of TDD, which is not an insignificant amount. It might be enough to make hacking it a thing players just automatically do, which leads to it becoming meta, which might be too far.

I've spent more time on this machine than most, and even something as simple as a single (not a stack) of medicated suture and mesh has people fearing that it becomes a greytider hacking meta type of item. This one has proven difficult to justify anything for, apple is the best I got.

Alpaccalypse replied:
Yeah, all this sounds right. I'm glad you are doing this and not me. I can see why all this needs to be thought through so thoroughly now. I hadn't even thought of these implication to what otherwise seems relatively innocuous.

Alpaccalypse reviewed (approved):
I've looked at this thoroughly, and been working on a lot of these inventories myself. There don't seem to be any mistakes, or notable balance issues. I've left a few comments that are just personal observations, but this gets my approval, for what it's worth.

If you decide to change anything I'll have another look and update this. 52 files looks daunting, but they're all just a few lines, and there's nothing here that is difficult to understand or check the specifics of.

AgentSmithRadio commented:
> Among other things, I was wondering why figurines even exist beyond hardcore RP content, this would answer that question. Someone mentioned bloat; this isn't bloat, it is making other game content no longer be bloat.

Figurines are a big point of contention because, well, they fill effectively every Drobe, which is a lot of machines, so they appear as the bulk of the content in this PR. I've written about them ad-nauseum, but in my mind, putting them in Drobes has 4 main purposes.

1. Figurines are fun. Toys are fun. People like toys in SS14, let them have toys, this is so explicity NT Branded that it could be considered a downright propaganda effort.

2. Figurines are inherently collectable, and there's no realistic way to collect them. I believe the current going rate for 10 random figurines is $4k? A full set is a DEDICATED cargo shift effort on a normal round, and even then, RNG is an absolute pain when it comes to this. Making them mostly accessible (Captain and HoP being map-spawn or somehow finding a CentDrobe dependent, outside of luck) makes the idea of doing a collection shift as a Librarian, Greytider, Chaplain or anything else suddenly becomes way more possible.

3. Figurines currently have a bounty in Cargo. This bounty isn't incompletable (it's no clown shoes + clown mask and wig), but it steals figurines from the rest of the station, denying other players avenues of fun. Hacking has limited use in Cargo as it is (usually for hacking out bounties such as the Beer or Wine bounty when there's no bartender, or getting seeds to plant in Botany), this gives another avenue for the department to complete the bounty without just having to buy a figurine crate.

4. It enables the 13-15 Figurine Thief objective to return. This objective sucked before. You either took the cash option from your thief kit and meta'd yourself to cargo to order a crate of figurines, or you'd raid the whole station and realize that you're short 5+ of them unless you were on a station like Cog. This makes the Figurine objective much more like the current Lamp objective. Lamp objective is trespassing into offices and stealing lamps, Figurine is about hacking machines and possibly trespassing to hack machines to acquire figurines. Unlike Lamps, there's some real silver tongue opportunities here, and it's a completely new gameplay angle for the Thief antagonist, and I'm dying to try it out.

Alpaccalypse commented:
I was thinking, half of these machines aren't going to be in immediately accessibly areas, so even if they can all be hacked, getting access to every department and possibly places like HoP office etc, is still quite an achievement to pull off.

AgentSmithRadio commented:
To keep this aging PR up to date, I've updated the inventories of clothesmate.yml to include https://github.com/space-wizards/space-station-14/pull/31710, dinnerware.yml to include https://github.com/space-wizards/space-station-14/pull/29318 and salvage.yml to include https://github.com/space-wizards/space-station-14/pull/33437. I've ignored the Smite Cranberry PRs to the various soda machines because those are rotating out shortly as per Christmas PRs.

Alpaccalypse commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/clothesmate.yml:
Looking at the comment here, I'm not actually aware of what uniform dying is, but might even more colors be an unwanted addition at this point, or is there a specific reason for this exception?

Boaz1111 replied:
this is from an already merged pr

Alpaccalypse replied:
Yep, I missed the comment above. All changes look good to me in that case.

SlamBamActionman commented:
> To keep this aging PR up to date, I've updated the inventories of clothesmate.yml to include #31710, dinnerware.yml to include #29318 and salvage.yml to include #33437. I've ignored the Smite Cranberry PRs to the various soda machines because those are rotating out shortly as per Christmas PRs.

The way you would do this is by merging the updated Master branch into your PR branch, not by manually editing the values yourself. Otherwise there's a risk you cause merge issues or accidentally edit out changes, so please fix this. 

Otherwise, I think this PR looks good!

AgentSmithRadio commented:
> The way you would do this is by merging the updated Master branch into your PR branch, not by manually editing the values yourself. Otherwise there's a risk you cause merge issues or accidentally edit out changes, so please fix this.

Understood, I'll get this corrected immediately, and do a full test.

AgentSmithRadio commented:
I reverted my offending commit, updated my master, rebased this branch with my master, tested every vending machine, realized that I reverted too far and corrected that issue. It required a force-push, but that's standard for rebasing. We should be good to go.

ArtisticRoomba commented on Resources/Prototypes/Catalog/VendingMachines/Inventories/seeds.yml:
```diff
@@ -39,5 +39,8 @@
     TowercapSeeds: 5
     WheatSeeds: 5
     WatermelonSeeds: 5
+  contrabandInventory:
+    BungoSeeds: 2
```
this could potentially be an issue, bungo seeds can be used to make bungotoxin

ArtisticRoomba reviewed (changes requested):
all looks good to me except that one thing

AgentSmithRadio commented:
Rebased to deal with merge conflict from [34458].

beck-thompson reviewed (approved):
Looks good!
